### ISSUE TITLE
`IsomorphismTransformationSemigroup` on an fp semigroup returns something of a different size

### ISSUE THREAD

**Issue by tomcontileslie** [semigroups/Semigroups#766]
As discussed with @james-d-mitchell.

The issue arises when running the following file, which defines a semigroup `fps`. (Can't seem to upload .g files so it's in .txt format). [problem-fp.txt](https://github.com/semigroups/Semigroups/files/6386339/problem-fp.txt)

Then, the following happens:
```
gap> Read("problem-fp.g");
#I  Assigned the global variables [ a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t, u, v, w, x, y, z, s1, s2, s3, s4, s5, s6, s7, s8,
  s9, s10, s11, s12, s13, s14, s15, s16, s17, s18, s19, s20, s21, s22, s23, s24 ]
gap> t;
<fp semigroup on the generators [ a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t, u, v, w, x, y, z,
  s1, s2, s3, s4, s5, s6, s7, s8, s9, s10, s11, s12, s13, s14, s15, s16, s17, s18, s19, s20, s21, s22, s23, s24 ]>
gap> Size(t);
159
gap> iso := IsomorphismTransformationSemigroup(fps);
MappingByFunction( <fp semigroup on the generators [ a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t,
  u, v, w, x, y, z, s1, s2, s3, s4, s5, s6, s7, s8, s9, s10, s11, s12, s13, s14, s15, s16, s17, s18, s19, s20,
  s21, s22, s23, s24 ]>, <transformation semigroup of size 159, degree 160 with 50 generators>
 , function( x ) ... end, function( x ) ... end )
gap> Size(Range(iso));
159
gap> Length(Elements(Range(iso)));
684
```
Note that this only happens if `t` is confirmed to be finite before running `IsomorphismTransformationSemigroup` - otherwise the range of the isomorphism has the correct size.

Saving the range as a list of generators and then re-defining the transformation semigroup from those generators yields a semigroup with size 684; presumably the size is only reported to be 159 because that number is assigned as an attribute at some point during the isomorphism creation process.

I'm afraid I stumbled across this example by chance so I'm not sure how to recreate a smaller one.

This output was obtained with GAP 4.10.2, Semigroups v3.4.2 (7fcd59b1), and libsemigroups v1.3.7 (0897d83). Hopefully it's reproducible for everyone else.

**Comment by wilfwilson** [comment]
(NB the file defines the fp semigroup as `t`, not as `fps`).
Thanks for the report.

In the case that the size is computed before creating the 'isomorphism', the result isn't a homomorphism, although it does seem to be injective.
```
gap> (t.1 * t.5) ^ iso = (t.1 ^ iso) * (t.5 ^ iso);
false
gap> Size(t) = Size(Set(t, x -> x ^ iso));
true
```

**Comment by wilfwilson** [comment]
GAP (without the Semigroups package loaded) seems to give the right answer - after about 10 minutes - so in this case the bug seems to be in Semigroups/libsemigroups. This is not surprising – I think we'd expect that to be the code that's running here.

**Comment by james-d-mitchell** [comment]
Resolved in v4.0.0
